pkg/service/repair/controller.go

```diff
@@ -202,25 +197,17 @@ func (c *rowLevelRepairController) shouldBlock(hosts []string, intensity float64
 		switch {
 		case intensity == maxIntensity:
 			ok = r <= c.limits[h].Max
-		case intensity <= 1:
-			ok = r <= c.limits[h].Default
 		default:
-			ok = r <= int(intensity)*c.limits[h].Default && r <= c.limits[h].Max
+			ok = r <= int(intensity) && r <= c.limits[h].Max
 		}
 		if !ok {
 			return false
 		}
 	}
 
-	// DENY if there is too much parallel replicas being repaired
-	if parallel := c.intensity.Parallel(); parallel != defaultParallel {
-		hash := replicaHash(hosts)
-		if _, ok := c.busyReplicas[hash]; !ok {
-			l := len(c.busyReplicas) + 1
-			if l > parallel || l > c.intensity.MaxParallel() {
-				return false
-			}
-		}
+	// DENY if there is too many repair tasks running
+	if parallel := c.intensity.Parallel(); parallel+1 > c.intensity.MaxParallel() {
```

Comment:
you stop considering busy replicas here
isn't parallel a static var?

Reply:
It turns out that this (and the condition above) was a total BS.
And one simply needs to compare `c.allJobs` (counter of currently running repair tasks) vs `c.intensity.Parallel()` - the actual value of `--parallel` for this repair task.